checkbox-provider-ce-oem/bin/multiple-otg.sh

```diff
@@ -0,0 +1,190 @@
+        mount -t configfs none "$(dirname "$gadget_path")" || exit 1
+    else
+        echo -e "\nInfo: configfs is already mounted!"
+        echo -e "\n$configfs"
+    fi
+    echo -e "\nInfo: Done!"
+}
+
+create_gadget() {
+    echo -e "\nInfo: Attempting to create gadget ..."
+    if [ ! -d "$gadget_path/g1" ]; then
+        mkdir "$gadget_path/g1" || exit 1
+    fi
+    if [ ! -d "$gadget_path/g1/strings/$language" ]; then
+        mkdir "$gadget_path/g1/strings/$language" || exit 1
+    fi
+    echo "$vid" > "$gadget_path/g1/idVendor" || exit 1
+    echo "$pid" > "$gadget_path/g1/idProduct" || exit 1
+    echo -e "\nInfo: Done!"
+}
+
+create_config() {
+    echo -e "\nInfo: Attempting to create config ..."
+    if [ ! -d "$gadget_path/g1" ]; then
+        create_gadget
+    fi
+    if [ ! -d "$gadget_path/g1/configs/c.1" ]; then
+        mkdir "$gadget_path/g1/configs/c.1" || exit 1
+    fi
+    echo 120 > "$gadget_path/g1/configs/c.1/MaxPower" || exit 1
+    echo -e "\nInfo: Done!"
+}
+
+create_function() {
+    echo -e "\nInfo: Attempting to create function ..."
+    local function=$1
+    modprobe "usb_f_$function" || exit 1
+    if [ ! -d "$gadget_path/g1/functions/$function.0" ]; then
+        mkdir "$gadget_path/g1/functions/$function.0" || exit 1
+    fi
+    if [ "$function" == "mass_storage" ]; then
+        tmp=$(mktemp -d)
+        dd if=/dev/zero of="$tmp/lun0.img" bs=1M count=16
+        mkdosfs -F 32 "$tmp/lun0.img"
+        echo "$tmp/lun0.img" > "$gadget_path/g1/functions/$function.0/lun.0/file" || exit 1
+    fi
+    ln -s "$gadget_path/g1/functions/$function.0" "$gadget_path/g1/configs/c.1" || exit 1
+    echo -e "\nInfo: Done!"
+}
+
+activate() {
+    echo -e "\nInfo: Attempting to activate function ..."
+    local udc=$1
+    echo "$udc" > "$gadget_path/g1/UDC" || exit 1
+    echo -e "\nInfo: Done!"
+}
+
+teardown() {
+    local function
+    echo -e "\nInfo: Attempting to teardown ..."
+    if [ -d "$gadget_path/g1" ]; then
+        echo "" > "$gadget_path/g1/UDC" || exit 1
+        for function in "$gadget_path/g1/configs/c.1/"*; do
+            function=$(basename "$function")
+            if [[ "$function" != "MaxPower" && "$function" != "bmAttributes" && "$function" != "strings" ]]; then
+                rm -f "$gadget_path/g1/configs/c.1/$function"
+                rmdir "$gadget_path/g1/functions/$function"
+            fi
+        done
+        rmdir "$gadget_path/g1/configs/c.1/"
+        rmdir "$gadget_path/g1/strings/$language"
+        rmdir "$gadget_path/g1"
+    else
+        echo -e "\nInfo: $gadget_path/g1 is not exist"
+    fi
+    echo -e "\nInfo: teardown done!"
+}
+
+
+main() {
+    if [[ -n "$udc" && -n "$function" ]]; then
+        init_gadget
+        create_gadget
+        create_config
+        create_function "$function"
+        activate "$udc"
+    fi
+    if [ "$teardown" == "true" ]; then
+        teardown
+    fi
+    if [[ -n "$config" ]]; then
```

Comment:
we could use if ... elif condition for these 3 purpose, since we will only run into one condition at once.